pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -2198,6 +2212,17 @@ pub fn remote_tenant_path(tenant_shard_id: &TenantShardId) -> RemotePath {
     RemotePath::from_string(&path).expect("Failed to construct path")
 }
 
+pub fn remote_tenant_manifest_path(
+    tenant_shard_id: &TenantShardId,
+    generation: Generation,
+) -> RemotePath {
+    let path = format!(
+        "tenants/{tenant_shard_id}/tenant-manifest.json{}",
+        generation.get_suffix()
```

Comment:
Probably, yeah. @jcsp why has this been done this way for `index-part.json`?
Would you do it differently now? The `.json` extension helps when you open it with an editor, etc.

Reply:
I think it was for search ergonomics in a situation where there were some without generations -- i.e. you could search for "index_part.json" and find all the indices, rather than having to search for index_part(-\d+)?.json or similar.

That motivation doesn't exist here, so it feels neater to tuck it before the `.`

Reply:
Oh, it's for S3 prefix listings.  So that we could list index_part.json* and get all of them.

Still, no need for that here, so let's put the generation before the dot.

Reply:
okay changed it to `tenant-manifest-01234.json`